android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java

```diff
@@ -0,0 +1,676 @@
+            }
+
+            globalParamRepository.saveGlobalParams(globalParamList);
+        } catch (Exception exception) {
+            Log.e(TAG, exception.getMessage(), exception);
+        }
+    }
+
+    private void parseToMap(Map<String, Object> map, Map<String, String> globalParamMap) {
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            String key = entry.getKey();
+
+            if (entry.getValue() instanceof HashMap) {
+                parseToMap((HashMap<String, Object>) entry.getValue(), globalParamMap);
+            } else {
+                globalParamMap.put(key, String.valueOf(entry.getValue()));
+            }
+        }
+    }
+
+    private Map<String, Object> getParams(String encodedCipher) {
+        try {
+            CryptoRequestDto cryptoRequestDto = new CryptoRequestDto();
+            cryptoRequestDto.setValue(encodedCipher);
+            CryptoResponseDto cryptoResponseDto = clientCryptoManagerService.decrypt(cryptoRequestDto);
+
+            byte[] data = CryptoUtil.base64decoder.decode(cryptoResponseDto.getValue());
+            Map<String, Object> paramMap = objectMapper.readValue(data, HashMap.class);
+            return paramMap;
+        } catch (IOException e) {
+            Log.e(TAG, "Failed to decrypt and parse config response >> ", e);
+        }
+        return Collections.EMPTY_MAP;
+    }
+
+    private void syncUserDetails(@NonNull SyncPigeon.Result<SyncPigeon.Sync> result) throws Exception {
+        Call<ResponseWrapper<UserDetailResponse>> call = syncRestService.fetchCenterUserDetails(
+                this.clientCryptoManagerService.getClientKeyIndex(), BuildConfig.CLIENT_VERSION);
+        call.enqueue(new Callback<ResponseWrapper<UserDetailResponse>>() {
+            @Override
+            public void onResponse(Call<ResponseWrapper<UserDetailResponse>> call, Response<ResponseWrapper<UserDetailResponse>> response) {
+                if (response.isSuccessful()) {
+                    ServiceError error = SyncRestUtil.getServiceError(response.body());
+                    if (error == null) {
+                        saveUserDetails(response.body().getResponse().getUserDetails());
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("UserDetailsSync")
+                                .setSyncProgress(Long.valueOf(3))
+                                .setErrorCode("")
+                                .build();
+                        result.success(sync);
+                        return;
+//                        Toast.makeText(context, "User Sync Completed", Toast.LENGTH_LONG).show();
+                    } else
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("UserDetailsSync")
+                                .setSyncProgress(Long.valueOf(3))
+                                .setErrorCode("UserDetailsSync Failed")
```

Comment:
This should be an errorCode not the errorMessage. Error code could be user_sync_failed and the message for this errorCode to be maintained in I10n files

Reply:
Noted. Will change the error message to error code.